src/PostgREST/Query/SqlFragment.hs

```diff
@@ -334,3 +335,12 @@ unknownLiteral = unknownEncoder . encodeUtf8
 intercalateSnippet :: ByteString -> [SQL.Snippet] -> SQL.Snippet
 intercalateSnippet _ [] = mempty
 intercalateSnippet frag snippets = foldr1 (\a b -> a <> SQL.sql frag <> b) snippets
+
+-- the "ctid" system column is always available to tables
+mutRangeF :: QualifiedIdentifier -> [FieldName] -> (SqlFragment, SqlFragment)
+mutRangeF mainQi rangeId = (
+    BS.intercalate " AND " $
+          (\col -> pgFmtColumn mainQi col <> " = " <> pgFmtColumn (QualifiedIdentifier mempty "pgrst_affected_rows") col) <$>
+          (if null rangeId then ["ctid"] else rangeId)
+  , if null rangeId then pgFmtColumn mainQi "ctid" else BS.intercalate ", " (pgFmtColumn mainQi <$> rangeId)
+  )
```

Comment:
Do I understand correctly that for a view, the `ctid` column will just be tried, too? And if it's not exposed, then the query will fail?

This is potentially dangerous, because a user could just create a view with a column named `ctid`, which does **not** select from the underlying table. Maybe it's not even unique. And then, the whole limited updates/delete feature would not work anymore - it would suddenly mutate more rows than requested.

Reply:
> Do I understand correctly that for a view, the ctid column will just be tried, too? And if it's not exposed, then the query will fail?

Yes, exactly that, I just let it fail.

> with a column named ctid, which does not select from the underlying table. Maybe it's not even unique. And then, the whole limited updates/delete feature would not work anymore - it would suddenly mutate more rows than requested.

Seems you cannot do that:

```
postgres=# create table hehe(ctid int primary key);
ERROR:  column name "ctid" conflicts with a system column name


```

Reply:
Also not for views:

```
postgres=# create table hehe_view as select 'hehe' as ctid, name from test.projects;
ERROR:  column name "ctid" conflicts with a system column name
```

Reply:
Argh, the above is for tables but for views it does work:

```sql
postgres=# create view hehe_view as select 'hehe' as ctid, name from test.projects;
CREATE VIEW
postgres=# table hehe_view ;
 ctid |    name
------+------------
 hehe | Windows 7
 hehe | Windows 10
 hehe | IOS
 hehe | OSX
 hehe | Orphan
(5 rows)
```

Reply:
Yes, just tried that, too.

So it seems we could extend the view parser to tell us whether a column is in-fact a ctid column on the base table. It wouldn't even matter which name it had in the view. That would be a much better solution, I think - although much more effort.

If you want to do something quick, I'd suggest to skip the `ctid` part for now?

Reply:
Hm, so what's the move for this one, remove the `ctid` fallback and only work for PKs?

Reply:
Leaving the `ctid` now that the feature only works for tables.